### PR TITLE
storage-blob-encryption-and-retention

### PR DESCRIPTION
**Pull request by tw3lveparsecs**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* added quickstart example storage-blob-encryption-and-retention

### CODE REVIEW

File: quickstarts/microsoft.storage/storage-blob-encryption-and-retention/azuredeploy.parameters.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "storageAccountName": {
+            "value": "GEN-UNIQUE"
+        },
+        "storageSku": {
```
Comment (bmoore-msft) [inline quickstarts/microsoft.storage/storage-blob-encryption-and-retention/azuredeploy.parameters.json:8]: move the literal param values to defaultValues in the template... that will make it easier to deploy via portal.

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.storage/storage-blob-encryption-and-retention/azuredeploy.parameters.json (hunk 1 above)
Comment (tw3lveparsecs) [reply quickstarts/microsoft.storage/storage-blob-encryption-and-retention/azuredeploy.parameters.json]: @bmoore-msft moved param values to default values as requested

**Comment by StephenWeatherford** [comment]
/azp run

NOTE: The bicep build error is an issue with the pipeline

File: quickstarts/microsoft.storage/storage-blob-encryption-and-retention/main.bicep [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  'BlockBlobStorage'
+])
+param storageKind string
+
+@description('Storage account access tier, Hot for frequently accessed data or Cool for infreqently accessed data')
+@allowed([
+  'Hot'
+  'Cool'
+])
+param storageTier string
```
Comment (bmoore-msft) [inline quickstarts/microsoft.storage/storage-blob-encryption-and-retention/main.bicep:35]: add defaultValue

File: quickstarts/microsoft.storage/storage-blob-encryption-and-retention/main.bicep [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+@description('Amount of days the soft deleted data is stored and available for recovery')
+@minValue(1)
+@maxValue(365)
+param deleteRetentionPolicy int
```
Comment (bmoore-msft) [inline quickstarts/microsoft.storage/storage-blob-encryption-and-retention/main.bicep:40]: add defaultValue

File: quickstarts/microsoft.storage/storage-blob-encryption-and-retention/main.bicep [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+@description('Enable blob encryption at rest')
+param blobEncryptionEnabled bool
```
Comment (bmoore-msft) [inline quickstarts/microsoft.storage/storage-blob-encryption-and-retention/main.bicep:43]: add defaultValue

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.storage/storage-blob-encryption-and-retention/main.bicep [hunk 5]
```diff
@@ -0,0 +1,74 @@
+@description('Storage account name')
+param storageAccountName string = 'GEN-UNIQUE'
```
[the remark this answers is not in the document]
Comment (tw3lveparsecs) [reply quickstarts/microsoft.storage/storage-blob-encryption-and-retention/main.bicep]: done

File: quickstarts/microsoft.storage/storage-blob-encryption-and-retention/main.bicep [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+@description('Storage account kind')
+@allowed([
+  'Storage'
+  'StorageV2'
+  'BlobStorage'
+  'FileStorage'
+  'BlockBlobStorage'
+])
+param storageKind string
```
[the remark this answers is not in the document]
Comment (tw3lveparsecs) [reply quickstarts/microsoft.storage/storage-blob-encryption-and-retention/main.bicep]: done

File: quickstarts/microsoft.storage/storage-blob-encryption-and-retention/main.bicep (hunk 3 above)
Comment (tw3lveparsecs) [reply quickstarts/microsoft.storage/storage-blob-encryption-and-retention/main.bicep]: done

File: quickstarts/microsoft.storage/storage-blob-encryption-and-retention/main.bicep (hunk 4 above)
Comment (tw3lveparsecs) [reply quickstarts/microsoft.storage/storage-blob-encryption-and-retention/main.bicep]: done

**Pull request merged**